### ISSUE TITLE
Only top half of 64x64 matrix displays

### ISSUE THREAD

**Issue by echamberlain** [pixelmatix/SmartMatrix_Fadecandy#3]
Hello,

I have a 64x64 matrix in a bottom to top C-shape configuration.  I think I've made all the appropriate changes, but when I run the fadecandy smartmatrix or process examples (modified for 64x64), only the top half (full width) of the matrix displays.

Here are my configuration changes:

fc_defs.h:

```
#define FRAME_WIDTH             64
#define FRAME_HEIGHT            64
```

SmartMatrix_Fadecandy.ino:

```
#define COLOR_DEPTH 24                  // In this sketch, COLOR_DEPTH only applies to scrollingLayer
const uint8_t kMatrixWidth = 64;        // known working: 32, 64, 96, 128
const uint8_t kMatrixHeight = 64;       // known working: 16, 32, 48, 64
const uint8_t kRefreshDepth = 36;       // known working: 24, 36, 48
const uint8_t kDmaBufferRows = 4;       // known working: 2-4, use 2 to save memory, more to keep from dropping frames and automatically lowering refresh rate
const uint8_t kPanelType = SMARTMATRIX_HUB75_32ROW_MOD16SCAN;   // use SMARTMATRIX_HUB75_16ROW_MOD8SCAN for common 16x32 panels
const uint8_t kMatrixOptions = (SMARTMATRIX_OPTIONS_C_SHAPE_STACKING | SMARTMATRIX_OPTIONS_BOTTOM_TO_TOP_STACKING);      // see http://docs.pixelmatix.com/SmartMatrix for options
const uint8_t kScrollingLayerOptions = (SM_SCROLLING_OPTIONS_NONE);
```

The build is configured for Teensy 3.2/3.1, No USB, 96 MHz optimized (overclock).

The build log:

```
Sketch uses 23,432 bytes (8%) of program storage space. Maximum is 262,144 bytes.
Global variables use 63,692 bytes (97%) of dynamic memory, leaving 1,844 bytes for local variables. Maximum is 65,536 bytes.
Low memory available, stability problems may occur.
```

Any idea why only the top half (full width) displays?

**Comment by echamberlain** [comment]
If I use the settings from SmartMatrix_Fadecandy.ino in FeatureDemo.ino, the full 64x64 matrix is used.

**Comment by echamberlain** [comment]
I found the problem.  I didn't update the fadecandy build to match the number of pixels in my display.

> The server is configured to send 2048 pixels - enough for a 32x64 pixel panel - per frame. You may get better performance by changing NUM_PIXELS in /server/src/smartmatrixdevice.h to the number of pixels in your panel.67

**Comment by embedded-creations** [comment]
How is the framerate at 64x64?  I found it to be quite slow.  You might try disabling interpolation to speed things up.  You'd have to change this code to use pixelNext without the call to lutInterpolate:
https://github.com/pixelmatix/SmartMatrix_Fadecandy/blob/master/Layer_Fadecandy.cpp#L121

**Comment by echamberlain** [comment]
I finally got a chance to revisit this project.  It took me a while to discover that a kRefreshDepth of 24 causes the display to remain blank after showing the fadecandy scroll.  I had to use a kRefreshDepth of 36.

I also decreased the kDmaBufferRows to 3.  With a setting of 4, I get low memory warnings.

Yes, the frame rate is a bit slow.  I get more scan line flicker than I would like.  Disabling the interpolation helped a bit.

I'm taking video from a FLIR Lepton infra-red module (80x60 pixels), it's refresh rate is only 10Hz, so I may have some room for improvement.
